src/nvim/screen.c

```diff
@@ -4569,20 +4549,62 @@ static bool use_cursor_line_sign(win_T *wp, linenr_T lnum)
          && (wp->w_p_culopt_flags & CULOPT_NBR);
 }
 
+// Return true if CursorLine highlight is to be used.
+static bool use_cursor_line(win_T *wp, linenr_T lnum, int row, int startrow, int filler_lines)
```

Comment:
Also this checks for whether `CursorLineNr` should be used, not whether `CursorLine` should be used.